sdk/scheduler/src/main/java/org/apache/mesos/specification/DefaultServiceSpecification.java

```diff
@@ -98,9 +131,13 @@ public String toJsonString() throws Exception {
     /**
      * Returns a {@link ConfigurationFactory} which may be used to deserialize
      * {@link DefaultServiceSpecification}s.
+     *
+     * @param additionalSubtypesToRegister any class subtypes which should be registered with
+     *     Jackson for deserialization. any custom placement rule implementations must be provided
      */
-    public static ConfigurationFactory<ServiceSpecification> getFactoryInstance() {
-        return FACTORY;
+    public static ConfigurationFactory<ServiceSpecification> getFactory(
```

Comment:
Added automatic validation (and unit tests to check that it works appropriately): Using the provided classes, serializes and deserializes the provided servicespecification. Then checks that the serialized/deserialized version equals() the original.

Validates:
- Any necessary classes have been provided up-front
- Equal()ity behaves consistently